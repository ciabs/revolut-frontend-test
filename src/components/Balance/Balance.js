import React from 'react';
import PropTypes from 'prop-types';

import {Container} from './Balance.styles';
import {getCurrencySymbol} from '../../helpers/currency.helper';

const Balance = ({currency, balance}) => (
    <Container>
      Balance: {getCurrencySymbol(currency)}{balance[currency].toFixed(2)}
    </Container>
  );

Balance.propTypes = {
  currency: PropTypes.string.isRequired,
  balance: PropTypes.object.isRequired
};

export default Balance;
