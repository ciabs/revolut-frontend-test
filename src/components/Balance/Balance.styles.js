import styled from 'styled-components';
import {balanceColor} from '../../styles/constants';

export const Container = styled.div`
  font-size: 14px;
  color: ${balanceColor};
`;
