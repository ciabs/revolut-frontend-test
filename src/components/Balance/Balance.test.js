import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Balance from './Balance';

describe('Balance', () => {
  const props = {
    balance: {
      EUR: 5
    },
    currency: 'EUR'
  };

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <Balance {...props} />,
      div
    );
  });

  test('has a valid snapshot', () => {
    const component = renderer.create(
      <Balance {...props} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
