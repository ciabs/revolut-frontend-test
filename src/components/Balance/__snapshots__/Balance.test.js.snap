// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Balance has a valid snapshot 1`] = `
<div
  className="sc-bdVaJa hEVXOE"
>
  Balance: 
  â‚¬
  5.00
</div>
`;
